Task4_Battleships/GameElements/Player.cs

```diff
@@ -30,31 +34,105 @@ public Player(string name)
+        public void PlacingShipsPhase()
         {
-            foreach (Ship ship in AvailableShips)
+            foreach (Ship ship in StartingShips)
             {
                 InputUtility.TakeShipInput();
+                AvailableShips.Add(ship);
                 PlaceSingleShip(ship, InputUtility.ShipCoordinates, InputUtility.IsVertical);
             }
         }
 
-        public void Strike(Tuple<char, int> coordinates)
+        public void Strike(GameBoard opponentBoard)
+        {
+            InputUtility.TakeStrikeInput();
+            foreach (BoardSquare square in opponentBoard.DefendingSide)
+            {
+                if (square.Coordinate.Equals(InputUtility.StrikeCoordinates))
+                {
+                    square.Strike();
+                    
+                    Board.AttackingSide.Find(sq => sq.Coordinate.Equals(square.Coordinate)).Strike();
+                }
+            }
+            // check for sunken ships -> update ship list
+        }
+
+        public void PlaceSingleShip(Ship ship, Tuple<int, int> coordinates, bool isVertical)
         {
-            // place pin on attacking board -> hit/miss/sink
+            if (!IsOverlappingShips(ship, coordinates, isVertical) || !IsOutOfBounds(ship, coordinates, isVertical))
+            {
+                foreach (BoardSquare sq in Board.DefendingSide)
+                {
+                    if (isVertical)
+                    {
+                        if (sq.Coordinate.Item1 == coordinates.Item1 && 
+                            sq.Coordinate.Item2 >=coordinates.Item2 || 
+                            sq.Coordinate.Item2 <= coordinates.Item2 + ship.Length)
+                        {
+                            sq.IsShip = true;
+                            sq.Icon = Convert.ToChar(ship.BoardLetter);
+                        }
+                    }
+                    else if (!isVertical)
+                    {
+                        if (sq.Coordinate.Item2 == coordinates.Item2 &&
+                            sq.Coordinate.Item1 >= coordinates.Item1 ||
+                            sq.Coordinate.Item1 <= coordinates.Item1 + ship.Length)
+                        {
+                            sq.IsShip = true;
+                            sq.Icon = Convert.ToChar(ship.BoardLetter);
+                        }
+                    }
+                }
+            }
+            else
+                throw new InvalidShipPlacementException("The ship is overlapping another one or is out of bounds!");
+        }
+
+        private bool IsOverlappingShips(Ship ship, Tuple<int, int> coordinates, bool isVertical)
+        {
+            if (isVertical)
+            {
+                var result = Board.DefendingSide
+                    .Select(sq => coordinates.Item1 == sq.Coordinate.Item1 && 
+                                 (coordinates.Item2 >= sq.Coordinate.Item2 || 
```

Comment:
(coordinates.Item2 >= sq.Coordinate.Item2 || coordinates.Item2 <= sq.Coordinate.Item2)
What are you trying to achieve with this condition?